pandas/tests/frame/test_missing.py

```diff
@@ -215,42 +215,46 @@ def test_dropna_tz_aware_datetime(self):
                              index=[0, 3])
         assert_frame_equal(result, expected)
 
-    def test_fillna(self):
-        tf = self.tsframe
+    def test_fillna_datetime(self, datetime_frame):
+        tf = datetime_frame
         tf.loc[tf.index[:5], 'A'] = np.nan
         tf.loc[tf.index[-5:], 'A'] = np.nan
 
-        zero_filled = self.tsframe.fillna(0)
+        zero_filled = datetime_frame.fillna(0)
         assert (zero_filled.loc[zero_filled.index[:5], 'A'] == 0).all()
 
-        padded = self.tsframe.fillna(method='pad')
+        padded = datetime_frame.fillna(method='pad')
         assert np.isnan(padded.loc[padded.index[:5], 'A']).all()
         assert (padded.loc[padded.index[-5:], 'A'] ==
                 padded.loc[padded.index[-5], 'A']).all()
 
-        # mixed type
-        mf = self.mixed_frame
-        mf.loc[mf.index[5:20], 'foo'] = np.nan
-        mf.loc[mf.index[-10:], 'A'] = np.nan
-        result = self.mixed_frame.fillna(value=0)
-        result = self.mixed_frame.fillna(method='pad')
-
         msg = "Must specify a fill 'value' or 'method'"
         with pytest.raises(ValueError, match=msg):
-            self.tsframe.fillna()
+            datetime_frame.fillna()
         msg = "Cannot specify both 'value' and 'method'"
         with pytest.raises(ValueError, match=msg):
-            self.tsframe.fillna(5, method='ffill')
+            datetime_frame.fillna(5, method='ffill')
+
+    def test_fillna_mixed_type(self, float_string_frame):
+
+        mf = float_string_frame
+        mf.loc[mf.index[5:20], 'foo'] = np.nan
+        mf.loc[mf.index[-10:], 'A'] = np.nan
+        result = float_string_frame.fillna(value=0)
+        result = float_string_frame.fillna(method='pad')  # noqa
```

Comment:
Rather than noqa can just remove assignment

Reply:
Well, unfortunately, this split (along fixtures) requested by @jreback is uncovering deficiencies in the existing test, in that *both* results are not tested for anything. Don't know what this should be...

Reply:
Sure entirely possible to have some mistakes in the existing tests. I'd suggest adding a TODO here to make a stronger assertion and opening a follow up issue. 

For time being OK to remove lhs and noqa note; at the very least I suppose this tests that the method doesn't raise